src/hidimstat/distilled_conditional_randomization_test.py

```diff
@@ -191,83 +184,59 @@ def fit(self, X, y):
         else:
             X_ = X
         y_ = y  # avoid modifying the original y
-        _, n_features = X_.shape
-
-        # screening process
-        if isinstance(self.estimator, Lasso) or isinstance(self.estimator, LassoCV):
-            ## Screening of variables for accelarate dCRT
-            if self.estimated_coef is None:
-                # base on the Theorem 2 of `liu2022fast`, the rule of screening
-                # is based on a cross-validated lasso
-                self.clf_screening_ = _fit_lasso(
-                    X_,
-                    y_,
-                    n_jobs=self.n_jobs,
-                    **self.params_lasso_screening,
+
+        if self.screening_threshold is not None:
+            if self.estimated_coef is not None:
+                warnings.warn(
+                    "Precomputed coefficients were provided, screening is skipped and "
+                    "screening_threshold is set to 100."
                 )
-                if hasattr(self.clf_screening_, "alpha_"):
-                    alpha_screening = self.clf_screening_.alpha_
-                else:
-                    alpha_screening = self.clf_screening_.alpha
-                # update the alpha value from the estimator on all values
-                self.params_lasso_screening["alpha"] = alpha_screening
-                if self.params_lasso_distillation_x is not None:
-                    self.params_lasso_distillation_x["alpha"] = alpha_screening
-                self.coefficient_ = np.ravel(self.clf_screening_.coef_)
+                self.selection_set = np.ones(X.shape[1], dtype=bool)
```

Comment:
```suggestion
                self.selection_set_ = np.ones(X.shape[1], dtype=bool)
```
`selection_set_` is an generated attribute; you should add an underline at this end. 
You should rename it in all the files.